src/addrman.cpp

```diff
@@ -1016,6 +1016,35 @@ std::optional<AddressPosition> AddrManImpl::FindAddressEntry_(const CAddress& ad
     }
 }
 
+bool AddrManImpl::HasFewReachableAddresses_(const std::unordered_set<Network>& networks)
```

Comment:
An outside node easily influences the outcome by feeding garbage, right? Then I'd suggest commenting along the following lines: "Should not be used for anything critical since it's easily [...]"

Reply:
Another improvement could be differentiating new/tried here, since tried is harder to influence.

Reply:
> An outside node easily influences the outcome by feeding garbage, right? 

I don't think so, since we're checking by network and we do not store addresses from unreachable networks.

Reply:
> Another improvement could be differentiating new/tried here, since tried is harder to influence.

Sure!